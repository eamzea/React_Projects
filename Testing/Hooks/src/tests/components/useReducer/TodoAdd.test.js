import "@testing-library/jest-dom";
import React from "react";
import { shallow } from "enzyme";
import TodoAdd from "../../../components/useReducer/TodoAdd";

describe("Pruebas en <TodoAdd/>", () => {
  const handelAddTodo = jest.fn(() => {});

  const wrapper = shallow(<TodoAdd handelAddTodo={handelAddTodo} />);

  test("Debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("No debe de llamar handelAddTodo", () => {
    const formSubmit = wrapper.find("Form").prop("onSubmit");

    formSubmit({ preventDefault() {} });

    expect(handelAddTodo).toHaveBeenCalledTimes(0);
  });

  test("Debe de llamar handelAddTodo", () => {
    const value = "Aprender React";
    wrapper.find("FormControl").simulate("onChange", {
      target: {
        value,
        name: "description",
      },
    });

    const newTodo = {
      id: new Date().getTime(),
      desc: value,
      done: false,
    };

    const formSubmit = wrapper.find("Form").prop("onSubmit");

    formSubmit({ preventDefault() {} });
    handelAddTodo(newTodo);

    expect(handelAddTodo).toHaveBeenCalledTimes(1);
    expect(handelAddTodo).toHaveBeenCalledWith(expect.any(Object));
    expect(handelAddTodo).toHaveBeenCalledWith({
      id: expect.any(Number),
      desc: value,
      done: false,
    });

    expect(wrapper.find("FormControl").prop("value")).toBe("");
  });
});
