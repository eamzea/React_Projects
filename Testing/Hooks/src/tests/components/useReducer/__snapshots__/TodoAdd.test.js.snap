// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoAdd/> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h4>
    Agregar To Do
  </h4>
  <hr />
  <Form
    inline={false}
    onSubmit={[Function]}
  >
    <InputGroup
      className="mb-1"
    >
      <FormControl
        aria-describedby="description"
        aria-label="description"
        name="description"
        onChange={[Function]}
        placeholder="Aprender..."
        type="text"
        value=""
      />
    </InputGroup>
    <Button
      active={false}
      block={true}
      disabled={false}
      type="submit"
      variant="outline-primary"
    >
      Agregar
    </Button>
  </Form>
</Fragment>
`;
